src/createContext.lua

```diff
@@ -61,8 +66,50 @@ local function createConsumer(context)
 		end
 	end
 
+	function Consumer:init(props)
+		-- This value may be nil, which indicates that our consumer is not a
+		-- descendant of a provider for this context item.
+		self.contextEntry = self:__getContext(context.key)
+	end
+
 	function Consumer:render()
-		return self.props.render(self.state.value)
+		-- Render using the latest available for this context item.
+		--
+		-- We don't store this value in state in order to have more fine-grained
+		-- control over our update behavior.
+		local value
+		if self.contextEntry ~= nil then
+			value = self.contextEntry.value
+		else
+			value = context.defaultValue
+		end
+
+		return self.props.render(value)
+	end
+
+	function Consumer:didUpdate()
+		-- Store the value that we most recently updated with.
+		--
+		-- This value is compared in the contextEntry onUpdate hook below.
+		self.lastValue = self.contextEntry.value
+	end
+
+	function Consumer:didMount()
+		if self.contextEntry ~= nil then
+			-- When onUpdate is fired, a new value has been made available in
+			-- this context entry, but we may have already updated in the same
+			-- update cycle.
+			--
+			-- To avoid sending a redundant update, we compare the new value
+			-- with the last value that we updated with (set in didUpdate) and
+			-- only update if they differ.
+			self.disconnect = self.contextEntry.onUpdate:subscribe(function(newValue)
```

Comment:
When can this happen?

Reply:
Updated the comment to make this more clear.

Reply:
Cool. Is it possible for our tests to cover this case?

Reply:
Good idea!

Reply:
Just added a test for this. I don't think it's directly related to this change, but is still a great test to have.